### PR TITLE
Refactor runtime-update/remove related code

### PR DESCRIPTION
**Pull request by bouwew**
All changes:
- In the platform files change to adding only the new devices: all at init, only new at runtime.
- Climate: return debug logging
- Coordinator: improve the detection of added/removed devices using `set()`, optimize code.
- Tests: rename the Tom to avoid the creation of double entities.

### CODE REVIEW

File: custom_components/plugwise/number.py [hunk 1]
```diff
@@ -76,7 +76,8 @@ def _add_entities() -> None:
             return
 
         entities: list[PlugwiseNumberEntity] = []
-        for device_id, device in coordinator.data.devices.items():
+        for device_id in coordinator.new_devices:
+            device = coordinator.data.devices[device_id]
```
[the remark this answers is not in the document]
Comment (bouwew) [reply custom_components/plugwise/number.py:80]: Ignore for pw-beta.

File: custom_components/plugwise/select.py [hunk 2]
```diff
@@ -83,7 +83,8 @@ def _add_entities() -> None:
             return
 
         entities: list[PlugwiseSelectEntity] = []
-        for device_id, device in coordinator.data.devices.items():
+        for device_id in coordinator.new_devices:
+            device = coordinator.data.devices[device_id]
```
[the remark this answers is not in the document]
Comment (bouwew) [reply custom_components/plugwise/select.py:87]: Ignore for pw-beta.

File: custom_components/plugwise/switch.py [hunk 3]
```diff
@@ -79,7 +79,8 @@ def _add_entities() -> None:
             return
 
         entities: list[PlugwiseSwitchEntity] = []
-        for device_id, device in coordinator.data.devices.items():
+        for device_id in coordinator.new_devices:
+            device = coordinator.data.devices[device_id]
```
[the remark this answers is not in the document]
Comment (bouwew) [reply custom_components/plugwise/switch.py:83]: Ignore for pw-beta.

File: custom_components/plugwise/binary_sensor.py [hunk 4]
```diff
@@ -97,13 +97,16 @@ async def async_setup_entry(
...
         """Add Entities."""
-        if not coordinator.new_devices:
+        if devices is None:
+            devices = coordinator.new_devices
+        if not devices:
             return
 
         entities: list[PlugwiseBinarySensorEntity] = []
-        for device_id, device in coordinator.data.devices.items():
+        for device_id in devices:
```
[the remark this answers is not in the document]
Comment (bouwew) [reply custom_components/plugwise/binary_sensor.py]: Update was reverted.

File: custom_components/plugwise/binary_sensor.py [hunk 5]
```diff
@@ -97,13 +97,16 @@
     coordinator = entry.runtime_data
 
     @callback
-    def _add_entities() -> None:
+    def _add_entities(devices: set[str] | None = None) -> None:
         """Add Entities."""
-        if not coordinator.new_devices:
+        if devices is None:
+            devices = coordinator.new_devices
+        if not devices:
```
[the remark this answers is not in the document]
Comment (bouwew) [reply custom_components/plugwise/binary_sensor.py]: Update was reverted.

File: custom_components/plugwise/binary_sensor.py [hunk 6]
```diff
@@ -124,7 +127,7 @@
 
     entry.async_on_unload(coordinator.async_add_listener(_add_entities))
 
-    _add_entities()
+    _add_entities(set(coordinator.data.devices))
```
[the remark this answers is not in the document]
Comment (bouwew) [reply custom_components/plugwise/binary_sensor.py]: Update was reverted.

**Review by CoMPaTech — APPROVED** [review]

**Pull request merged**